### ISSUE TITLE
Update statistics in HTML tables when filtering rows

### ISSUE THREAD

**Issue by PhilippWendler** [sosy-lab/benchexec#125]
The statistics in the footer of the HTML tables should get automatically recomputed after the row filter was changed. For this the raw data values need to be added to the HTML, possibly as additional `data` attributes to the table cell tags.

The statistics computation inside the table-generator should still stay such that the statistics are also available if JavaScript is blocked in the browser, and to not delay loading the table further.

**Comment by PhilippWendler** [comment]
There is the question how to show the filtered statistics, there are two possibilities:

1. If a filter is defined, *replace everything* that is shown in the statistics with statistics over the filtered data, and replace the text `total results` with something like `filtered tasks`. The row `local summary` would be hidden in this case (because it is not possible to show filtered data in it.)
    *Advantage*: Consistent with other tabs (table view, plots) which also have this behavior of using only the filtered data.
    *Disadvantage*: Potentially unexpected for long-time users of BenchExec that expect statistics to be unfiltered (could confuse filtered with non-filtered statistics). Only one kind of statistics (filtered/unfiltered) shown at a given time.
2. If a filter is defined, insert an *additional row* with the statistics data over the filtered data (without breakdown into correct/incorrect etc.) and keep all the other rows as they are:
    ![grafik](https://user-images.githubusercontent.com/2545335/152316960-df0e9a1f-8d2d-4f66-b7cf-97fa536553a7.png)
    *Advantage*: Does not confuse long-time users, shows both filtered and unfiltered statistics at the same time.
    *Disadvantage*: No breakdown of filtered statistics into correct/incorrect etc.

[Previously](https://github.com/sosy-lab/benchexec/pull/616#pullrequestreview-638682810) I was in favor of option 2, but now I am not so sure anymore. Option 1 does seem more consistent and intuitive overall (disregarding the old behavior of the statistics table), so maybe it is the better long-term solution.

Note that the other arguments about which statistics are visible are not so critical: For Option 1, users can easily toggle between filtered and unfiltered statistics by deleting the filters (2 clicks) and resurrecting them (back button). There has also never been a request for seeing unfiltered plots while filters are defined. For Option 2, users can manually request the desired statistics with additional filters, though if you want to get them for many run sets, this can involve some clicking.

Could you please give feedback?
Vote :rocket: for Option 1 and :tada: for Option 2. Of course, also feel free to reply with a comment.

**Comment by dbeyer** [comment]
I am in favor of option 1.

But what I do not understand is the difference between total/filtered `results` vs. `tasks`.
Is a result a row that is different from "empty", and/or different from "--"?
Perhaps use `total rows` and `filtered rows`?

Independently, I also suggest to change "filtered" to "selected".  Because if we show 10 rows out of a total of 100 rows, then we actually have **filtered** (i.e., applied filtering to) the *100 rows* and this results in **10 selected rows**.
It might not be clear whether `filtered` refers to the set that was filtered, or the result of the filtering process.
Even worse, I can even think of a notion of `filtered out rows` (are they the selected or the not selected?).

**Comment by PhilippWendler** [comment]
Yes, a result is a non-empty table entry. There can be 100 rows/tasks, but for example only 10 results in total for a specific tool. The wording "total results" was just recently added based on your suggestion in order to make this clear.
"total rows" would be wrong because the numbers in this row refer to the number of results for each tool, not the number of table rows/tasks. More specifically, the numbers in the status column tell how many results the respective tool has.

Using "selected" is fine.

**Comment by PhilippWendler** [comment]
Maybe replacing "total results" with "all results" would be good, and switching to "all results on selected tasks" when a filter is applied.

**Comment by TBunk** [comment]
How about going with option 1 and appending the overall result again below there, maybe either blurred or as and expandable menu that can be (un-)folded? That way the user would have both available and could also more easily compare the filtered results against the overall results.

**Comment by PhilippWendler** [comment]
That would be in principle be doable. But comparing filtered/unfiltered statistics would be (somewhat) doable by using the browser's back/forward buttons anyway, and we never had a similar request for other views (table, plots). So I would prefer to not add this until there is an explicit feature request (it is easy to add later).

**Comment by lembergerth** [comment]
I'm also in favor of option 1.
I'm regularly confused by this; everything (detailed results, plots) is showing the selected data, but the overview is not.

If one wants to see the overview of all data, the filter can be removed (just like with all the other features that work on filtered data).

> But comparing filtered/unfiltered statistics would be (somewhat) doable by using the browser's back/forward buttons anyway, and we never had a similar request for other views (table, plots).

I just use two browser windows or tabs if I want to compare data across selections.

**Comment by PhilippWendler** [comment]
I guess we can simply suffix each of the names of the statistics rows with ` of selected tasks` if there is a filter, i.e.,
```
all results of selected tasks
  correct results of selected tasks
    correct true of selected tasks
    ...
```
That should make it quite visible that the meaning of the statistics has changed compared to previous versions.
